test/format/verify.c

```diff
@@ -167,13 +167,28 @@ table_verify_mirror(WT_CONNECTION *conn, TABLE *base, TABLE *table, const char *
         testutil_check(__wt_snprintf(buf, sizeof(buf), "checkpoint=%s", checkpoint));
 
     /*
-     * If opening a checkpoint, retry if the cursor checkpoint IDs don't match, it just means that a
-     * checkpoint happened between the two open calls.
+     * If opening a checkpoint, retry if:
+     * - one of the cursors could not be opened or,
+     * - the cursor checkpoint IDs don't match, it just means that a checkpoint happened between the
+     * two open calls.
      */
     for (;;) {
         wt_wrap_open_cursor(session, base->uri, checkpoint == NULL ? NULL : buf, &base_cursor);
-        base_id = base_cursor->checkpoint_id(base_cursor);
         wt_wrap_open_cursor(session, table->uri, checkpoint == NULL ? NULL : buf, &table_cursor);
+
+        /* Check both cursors were successfully opened. */
+        if (checkpoint != NULL && (base_cursor == NULL || table_cursor == NULL)) {
```

Comment:
We don't try to reopen them, we close the one that might have been successfully opened while the other failed.
If this check passes, we expect both cursors to be opened, hence the `WT_ASSERT`.

Reply:
I guess I don't follow then... as in my understanding the `continue` statement after you close the open cursor will restart the loop where you try to open both cursors.

Reply:
Also how will you exit this loop if checkpoint != NULL?

Reply:
> as in my understanding the continue statement after you close the open cursor will restart the loop where you try to open both cursors

I at least one call keeps failing, we will loop indefinitely. @sueloverso, do you agree this can happen? Should we try for a number of times and skip validation or error out if this retries too many times?

Reply:
I'm trying to understand the ramifications of this change. Right now format does a bulk load on all tables and then does a verify after all the tables have their initial population complete. The workload and checkpoint threads are not yet but the code should guarantee that a system wide checkpoint has happened. The verify code is called after the bulk load but `wts_verify` does an unconditional checkpoint before checking. The other caller that might verify mirrors is from `wts_verify_checkpoint` which is called from the checkpoint thread after completing a system-wide checkpoint. Am I forgetting a different path for how we get here?

Reply:
You are right and we are trying to address the path from `wts_verify_checkpoint`. When we want to verify a checkpoint and get here.

Reply:
After discussing with @ajmorton, we have decided to move the logic into `wt_wrap_open_cursor` to retry in case of a failure when opening a checkpoint cursor instead of expecting the caller to handle it.